### PR TITLE
Added Junit and integration test cases and endpoint to fetch all data…

### PR DESCRIPTION
**Pull request by Arundhathy-k**
Added Junit and Integration test cases and endpoint to fetch all data from mongo db.

### CODE REVIEW

File: newsService/pom.xml [hunk 1]
```diff
@@ -41,6 +41,7 @@
             <version>0.0.1-SNAPSHOT</version>
             <scope>compile</scope>
         </dependency>
+
```
Comment (dhanyesh) [inline newsService/pom.xml]: I don't want any changes in this file. please revert it back.

File: newsService/src/main/resources/application.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
 server.port = 8081
 spring.application.name = newsService
-logging.level.org.springframework.web: DEBUG
```
Comment (dhanyesh) [inline newsService/src/main/resources/application.properties]: Why you removed this property ? explain

File: pom.xml [hunk 3]
```diff
@@ -28,6 +28,7 @@
 		<module>notificationService</module>
 		<module>mysql-dao</module>
 		<module>mongo-dao</module>
+		<module>newsService</module>
```
Comment (dhanyesh) [inline pom.xml]: why this change here in this task ? Shouldn't it be part of the skeleton code?

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: newsService/src/main/resources/application.properties (hunk 2 above)
Comment (Arundhathy-k) [reply newsService/src/main/resources/application.properties]: added property.

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java [hunk 4]
```diff
@@ -14,11 +15,13 @@ public class NewsServiceController {
     @Autowired
     private NewsService newsService;
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java]: Use constructor injection

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java [hunk 5]
```diff
@@ -14,11 +15,13 @@ public class NewsServiceController {
...
 
-
-    @GetMapping("/top-headlines")
+    @PostMapping("/top-headlines")
```
Comment (dhanyesh) [inline newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java]: change endpoint don't use - in name

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java (hunk 5 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java]: Changed end point.

File: newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java (hunk 4 above)
Comment (Arundhathy-k) [reply newsService/src/main/java/com/kovan/app/controller/NewsServiceController.java]: used constructor injection.

File: newsService/src/test/java/com/kovan/unit/NewsServiceTests.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import org.springframework.web.client.RestTemplate;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class NewsServiceTests {
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/unit/NewsServiceTests.java]: change to NewsServiceTest

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import java.util.List;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.springframework.boot.test.context.SpringBootTest.WebEnvironment.RANDOM_PORT;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = RANDOM_PORT)
+public class NewsServiceControllerIT {
```
Comment (dhanyesh) [inline newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java]: NewsServiceIT

File: mongo-dao/src/main/java/com/kovan/mapper/NewsMapper.java [hunk 8]
```diff
@@ -4,8 +4,6 @@
 import com.kovan.entity.NewsEntity;
 import io.micrometer.common.util.StringUtils;
 import org.springframework.stereotype.Component;
-
-import java.time.LocalDateTime;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
```
Comment (dhanyesh) [inline mongo-dao/src/main/java/com/kovan/mapper/NewsMapper.java:9]: In this class you have used StringUtils can you please make it as static import.

**Review by dhanyesh — CHANGES_REQUESTED** [review]

File: newsService/src/test/java/com/kovan/unit/NewsServiceTests.java (hunk 6 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/unit/NewsServiceTests.java]: changed

File: newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java (hunk 7 above)
Comment (Arundhathy-k) [reply newsService/src/test/java/com/kovan/integration/NewsServiceControllerIT.java]: Changed

File: mongo-dao/src/main/java/com/kovan/mapper/NewsMapper.java (hunk 8 above)
Comment (Arundhathy-k) [reply mongo-dao/src/main/java/com/kovan/mapper/NewsMapper.java:9]: made import as static.
